agent/job_runner.go

```diff
@@ -250,6 +250,17 @@ func (r *JobRunner) Run() error {
 
 	startedAt := time.Now()
 
+	// Publish metric for how long this job was in the queue for, if we can
+	// calculate that
+	if r.job.RunnableAt != "" {
+		runnableAt, err := time.Parse(time.RFC3339Nano, r.job.RunnableAt)
+		if err != nil {
+			r.logger.Error("Metric submission failed to parse %s", r.job.RunnableAt)
+		} else {
+			r.metrics.Timing("queue.duration", startedAt.Sub(runnableAt))
+		}
+	}
+
```

Comment:
Does it feel appropriate to put this just before we make the "start job" request? Should it be after we "start job" on the Buildkite API?

Reply:
Good thinking! I’ve been looking at this and agree that the metric should be submitted after `startJob` returns success.

A case I thought of where startJob could fail, is when multiple agents are started with `--acquire-job` for a given job’s UUID. In the event an agent fails to grab the job because another agent has already grabbed it, the latter agent’s start time isn’t the one that matters.

I’ll make that change now and merge this so it’s included in tomorrow’s release. 🎉